android/src/app/organicmaps/editor/AdvancedTimetableFragment.java

```diff
@@ -125,19 +130,30 @@ void setTimetableChangedListener(@NonNull TimetableChangedListener listener)
     setTextChangedListener(mInput, mListener);
   }
 
-  private static void setTextChangedListener(@Nullable EditText input,
-                                             @Nullable TimetableChangedListener listener)
+  private void setTextChangedListener(@Nullable EditText input,
+                                      @Nullable TimetableChangedListener listener)
   {
     if (input == null || listener == null)
       return;
-
     input.addTextChangedListener(new TextWatcher()
     {
       @Override
       public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
 
       @Override
-      public void onTextChanged(CharSequence s, int start, int before, int count) {}
+      public void onTextChanged(CharSequence s, int start, int before, int count)
+      {
+        boolean isValidFormat = OpeningHours.nativeIsTimetableStringValid(s.toString());
+        mSaveButton.setEnabled(isValidFormat);
+        if (!isValidFormat)
+        {
+          mSaveButton.setColorFilter(Color.argb(150, 200, 200, 200));
+        }
+        else
+        {
+          mSaveButton.setColorFilter(null);
+        }
```

Comment:
```suggestion
        mSaveButton.setColorFilter(isValidFormat ? null : Color.argb(150, 200, 200, 200));
```

Reply:
For some reason, this change makes the app crash while launching the Advanced timetable.

Reply:
We are engineers, there is no "some reason", it should be a clear error message ;-)

Reply:
Isn't the code I proposed identical to yours?

Reply:
It throws this error:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
```
If I replace the null with Color.TRANSPARENT like this It does not crash: 
```
mSaveButton.setColorFilter(isValidFormat ? Color.TRANSPARENT : Color.argb(150, 200, 200, 200));
```

Reply:
It is not clear why it works in your version because there is no setColorFilter(int) method.

On the other side, the button can be changed in the future, and hardcoding here the color value doesn't look like a flexible solution. It would be great to have only enable/disable call in the code, and specify colors for enabled/disabled state in resources, for example, like it is mentioned here: https://stackoverflow.com/a/33406917

Reply:
The link you mentioned used textColor which cannot be used with our implementation since we are using an imageview as a button.
But there is `app:tint` which does the same thing.
I just updated my implementation of it have a look at it & let me know if there is anything to change there.
Here are some screenshots from the new implementation (changed the shade of grey this one looks much better in my opinion than the older one).
Enabled             |  Disabled
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/46300244/226186110-8e4e7ce9-6146-4c87-9f86-8186a5698e3d.png) | ![image](https://user-images.githubusercontent.com/46300244/226186099-48cf24c7-790f-491a-90dc-e0e230a9abb4.png)

_Will sign & rebase the commits if you approve of this implementation :)_